mne_connectivity/decoding/decomposition.py

```diff
@@ -0,0 +1,472 @@
+        else:
+            csd_kwargs.update(
+                {"frequencies": self.cwt_freqs, "n_cycles": self.cwt_n_cycles}
+            )
+            csd = csd_array_morlet(**csd_kwargs)
+
+        if self.mode in ["multitaper", "fourier"]:
+            fmin = self.fmin
+            fmax = self.fmax
+        else:
+            fmin = self.cwt_freqs[0]
+            fmax = self.cwt_freqs[-1]
+        csd = csd.sum(fmin, fmax).get_data(index=0)
+        csd = np.reshape(csd, csd.shape[0] ** 2)
+
+        return np.expand_dims(csd, 1)
+
+    def _extract_filters_and_patterns(self):
+        """Extract filters and patterns from the connectivity estimator."""
+        # XXX: need to sort indices and transpose patterns when multiple comps supported
+        self.filters_ = (
+            self._conn_estimator.filters[0, 0, : len(self.indices[0]), 0],
+            self._conn_estimator.filters[1, 0, : len(self.indices[1]), 0],
+        )
+
+        self.patterns_ = (
+            self._conn_estimator.patterns[0, 0, : len(self.indices[0]), 0],
+            self._conn_estimator.patterns[1, 0, : len(self.indices[1]), 0],
+        )
+
+        # XXX: remove once support for multiple comps implemented
+        self.filters_ = (
+            np.expand_dims(self.filters_[0], 1),
+            np.expand_dims(self.filters_[1], 1),
+        )
+        self.patterns_ = (
+            np.expand_dims(self.patterns_[0], 0),
+            np.expand_dims(self.patterns_[1], 0),
+        )
+
+    def transform(self, X):
+        """Decompose data into connectivity sources using the fitted filters.
+
+        Parameters
+        ----------
+        X : array, shape=((n_epochs, ) n_signals, n_times)
+            The data to be transformed by the connectivity decomposition filters.
+
+        Returns
+        -------
+        X_transformed : array, shape=((n_epochs, ) n_components*2, n_times)
+            The transformed data. The first ``n_components`` channels are the
+            transformed seeds, and the last ``n_components`` channels are the
+            transformed targets.
+        """
+        self._check_X(X, ndim=(2, 3))
+        if self.filters_ is None:
+            raise RuntimeError(
+                "no filters are available, please call the `fit` method first"
+            )
+
+        # transform seed and target data
+        X_seeds = self.filters_[0].T @ X[..., self.indices[0], :]
+        X_targets = self.filters_[1].T @ X[..., self.indices[1], :]
+
+        return np.concatenate((X_seeds, X_targets), axis=-2)
+
+    def fit_transform(self, X, y=None, **fit_params):
+        """Fit filters to data, then transform and return it.
+
+        Parameters
+        ----------
+        X : array, shape=(n_epochs, n_signals, n_times)
+            The input data which the connectivity decomposition filters should be fit to
+            and subsequently transformed.
+        y : None
+            Used for scikit-learn compatibility.
```

Comment:
[nitpick] can this be a bit clearer? I'm assuming the param needs to be in the signature for this class to be usable in sklearn pipelines; if that's right maybe say: "Ignored; exists for compatibility with scikit-learn pipelines" ?

Reply:
Yeah, I think it's worth propagating that docstring change upstream

Reply:
Okay, submitted a PR: https://github.com/mne-tools/mne-python/pull/12664